human_aware_rl/imitation/behavior_cloning_tf2.py

```diff
@@ -0,0 +1,358 @@
+import os, pickle, copy
+from tensorflow import keras
+import tensorflow as tf
+import numpy as np
+from human_aware_rl.human.process_dataframes import get_trajs_from_data
+from human_aware_rl.rllib.rllib import RlLibAgent, softmax, evaluate, get_base_env, get_mlp
+from overcooked_ai_py.mdp.actions import Action
+from overcooked_ai_py.agents.agent import AgentPair
+from overcooked_ai_py.agents.benchmarking import AgentEvaluator
+from overcooked_ai_py.mdp.overcooked_env import OvercookedEnv
+from overcooked_ai_py.mdp.overcooked_mdp import OvercookedGridworld
+from overcooked_ai_py.planning.planners import MediumLevelPlanner, NO_COUNTERS_PARAMS
+from overcooked_ai_py.mdp.overcooked_env import DEFAULT_ENV_PARAMS
+from ray.rllib.policy import Policy as RllibPolicy
+from human_ai_msr.data_dir import DATA_DIR
+
+#################
+# Configuration #
+#################
+
+HUMAN_DATA_PATH = os.path.join(DATA_DIR, "clean_train_trials.pkl")
+BC_SAVE_DIR = os.path.join(DATA_DIR, "bc_runs")
+
+DEFAULT_DATA_PARAMS = {
+    "train_mdps": ["cramped_room"],
+    "ordered_trajs": False,
+    "human_ai_trajs": False,
+    "processed" : True,
+    "data_path": HUMAN_DATA_PATH
+}
+
+DEFAULT_MLP_PARAMS = {
+    # Number of fully connected layers to use in our network
+    "num_layers" : 2,
+    # Each int represents a layer of that hidden size
+    "net_arch" : [64, 64]
+}
+
+DEFAULT_TRAINING_PARAMS = {
+    "epochs" : 100,
+    "validation_split" : 0.15,
+    "batch_size" : 64,
+    "learning_rate" : 1e-3,
+    "use_class_weights" : False
+}
+
+DEFAULT_EVALUATION_PARAMS = {
+    "ep_length" : 400,
+    "num_games" : 1,
+    "display" : False
+}
+
+DEFAULT_BC_PARAMS = {
+    "data_params": DEFAULT_DATA_PARAMS,
+    "mdp_params": {'layout_name': "cramped_room"},
+    "env_params": DEFAULT_ENV_PARAMS,
+    "mdp_fn_params": {},
+    "mlp_params" : DEFAULT_MLP_PARAMS,
+    "training_params" : DEFAULT_TRAINING_PARAMS,
+    "evaluation_params" : DEFAULT_EVALUATION_PARAMS
+}
+
+def _get_base_env(bc_params):
+    return get_base_env(bc_params['mdp_params'], bc_params['env_params'])
+
+def _get_observation_shape(bc_params):
+    """
+    Helper function for creating a dummy environment from "mdp_params" and "env_params" specified
+    in bc_params and returning the shape of the observation space
+    """
+    base_env = _get_base_env(bc_params)
+    dummy_state = base_env.mdp.get_standard_start_state()
+    mlp = get_mlp(base_env)
+    obs_shape = base_env.mdp.featurize_state(dummy_state, mlp)[0].shape
+    return obs_shape
+
+DEFAULT_BC_PARAMS["action_shape"] = (len(Action.ALL_ACTIONS), )
```

Comment:
Revised this section to lazily load the default bc params with a function, instead of loading them every time the script is executed (something I meant to do early). I think that's a pretty solid convention and it's something I've seen used a lot in other python code

Reply:
Yup, sounds good!